### DISCUSSION TITLE
What exactly is the relation between Kind and Kindelia?

### DISCUSSION THREAD

**Discussion by ken-okabe** [uwu-tech/Kind#316] [Q&A] [answered]
I'm new to Kind.
I'm also very interested in [Kindelia
](https://github.com/Kindelia/Kindelia).

I've read
Kindelia: a minimal decentralized computer
is designed for the extensive purpose of virtual currency and I expect eventually, the engine/VM will be substituted to [HOVM](https://github.com/Kindelia/HOVM).

What I wonder is what exactly is the relation between Kind and Kindelia?
Kindelia documentation is full of the virtual currency topic, and not much about Syntax, etc...

So here is my imagination for now.

Kindelia = Kind + extensive feature for virtual currency.

Or Kind will be ported as the subset to Kinderia?

Please let me know, thanks!

**Comment by VictorTaelin** [comment, marked as answer]
Thanks for the questions.

Kindelia is basically a "functional Ethereum", that replaces the clunky EVM by a high-order machine (which I'm currently calling HOVM). Note that the paper of Kindelia is outdated. It is a draft that was written before HOVM was designed, so it will soon be rewritten and the syntax updated.

Kind is a proof and programming language based on self encodings. Kind will compile to the HOVM, which means you'll be able to use Kind to write smart-contracts in Kindelia. This will be specially nice since functional languages tend to be way more secure than imperative languages, and Kindelia is a **proof** language, so formally verifying a contract will be very easy. Security is very important for smart-contracts, since they can't be patched, so Kind -> HOVM -> Kindelia is a powerful combination.

Note that Kindelia does NOT depend on Kind. In theory you could compile other languages to Kindelia, specially if they're functional. Haskell for example is very similar to HOVM's language, so deploying pure Haskell contracts to Kindelia shouldn't be hard.

**Comment by ken-okabe** [comment]
Thank you for the extensive explanation. Now everything is so clear to me.
Really appreciated. Well designed, so impressive projects!
